--- a/drivers/pinctrl/ralink/pinctrl-mt7621.c
+++ b/drivers/pinctrl/ralink/pinctrl-mt7621.c
@@ -39,12 +39,12 @@ static struct ralink_pmx_func i2c_grp[] 
 static struct ralink_pmx_func uart3_grp[] = {
 	FUNC("uart3", 0, 5, 4),
 	FUNC("i2s", 2, 5, 4),
-	FUNC("spdif3", 3, 5, 4),
+	FUNC("spdif3", 3, 5, 1),
 };
 static struct ralink_pmx_func uart2_grp[] = {
 	FUNC("uart2", 0, 9, 4),
 	FUNC("pcm", 2, 9, 4),
-	FUNC("spdif2", 3, 9, 4),
+	FUNC("spdif2", 3, 11, 1),
 };
 static struct ralink_pmx_func jtag_grp[] = { FUNC("jtag", 0, 13, 5) };
 static struct ralink_pmx_func wdt_grp[] = {
